### PR TITLE
Upgrade dependencies for snyk check

### PR DESCRIPTION
**Pull request by aprudhomme**
Did a pass with app.snyk.io to check for vulnerabilities in dependencies, then fixed what I could.

I also removed `io.grpc:grpc-testing` from the clientlib api. It is only used for testing and should not be included in the runtime classpath. This will probably require adding an extra dependency for plugin tests.

## Remaining Issues
```
Server:
* com.typesafe.akka:akka-http-core_2.12@10.1.12: upgrade to 10.2.7 for
      Improper Resource Shutdown or Release [high severity][https://snyk.io/vuln/SNYK-JAVA-COMTYPESAFEAKKA-2336361]
  * org.eclipse.jetty:jetty-io@9.4.19.v20190610: upgrade to 9.4.39.v20210325 for
      Denial of Service (DoS) [high severity][https://snyk.io/vuln/SNYK-JAVA-ORGECLIPSEJETTY-1090340]
  * com.squareup.okio:okio@1.17.5: upgrade to 3.4.0 for
      Denial of Service (DoS) [medium severity][https://snyk.io/vuln/SNYK-JAVA-COMSQUAREUPOKIO-5773320]
  * com.typesafe.akka:akka-actor_2.12@2.5.31: upgrade to 2.8.1 for
      Generation of Predictable Numbers or Identifiers [medium severity][https://snyk.io/vuln/SNYK-JAVA-COMTYPESAFEAKKA-5518120]
  * org.eclipse.jetty:jetty-server@9.4.19.v20190610: upgrade to 9.4.37.v20210219 for
      Denial of Service (DoS) [medium severity][https://snyk.io/vuln/SNYK-JAVA-ORGECLIPSEJETTY-1080611]
  * org.eclipse.jetty:jetty-server@9.4.19.v20190610: upgrade to 9.4.51 for
      Denial of Service (DoS) [medium severity][https://snyk.io/vuln/SNYK-JAVA-ORGECLIPSEJETTY-5426159]
  * org.eclipse.jetty:jetty-http@9.4.19.v20190610: upgrade to 9.4.52.v20230823 for
      Improper Handling of Length Parameter Inconsistency [medium severity][https://snyk.io/vuln/SNYK-JAVA-ORGECLIPSEJETTY-5902998]
  * com.typesafe.akka:akka-http-core_2.12@10.1.12: upgrade to 10.2.4 for
      HTTP Request Smuggling [medium severity][https://snyk.io/vuln/SNYK-JAVA-COMTYPESAFEAKKA-2315411]
  * org.eclipse.jetty:jetty-server@9.4.19.v20190610: upgrade to 9.4.35.v20201120 for
      HTTP Request Smuggling [medium severity][https://snyk.io/vuln/SNYK-JAVA-ORGECLIPSEJETTY-1047304]
  * com.typesafe.akka:akka-http_2.12@10.1.12: upgrade to 10.5.2 for
      Creation of Temporary File With Insecure Permissions [medium severity][https://snyk.io/vuln/SNYK-JAVA-COMTYPESAFEAKKA-5596911]
  * org.eclipse.jetty:jetty-server@9.4.19.v20190610: upgrade to 11.0.3 for
      Information Exposure [low severity][https://snyk.io/vuln/SNYK-JAVA-ORGECLIPSEJETTY-1313686]
  * org.eclipse.jetty:jetty-http@9.4.19.v20190610: upgrade to 9.4.47 for
      Improper Input Validation [low severity][https://snyk.io/vuln/SNYK-JAVA-ORGECLIPSEJETTY-2945452]
  * org.eclipse.jetty:jetty-server@9.4.19.v20190610: upgrade to 9.4.51 for
      Information Exposure [low severity][https://snyk.io/vuln/SNYK-JAVA-ORGECLIPSEJETTY-5426160]
  * org.eclipse.jetty:jetty-http@9.4.19.v20190610: upgrade to 9.4.51 for
      Information Exposure [low severity][https://snyk.io/vuln/SNYK-JAVA-ORGECLIPSEJETTY-5426161]

Clientlib:
  * com.squareup.okio:okio@1.17.5: upgrade to 3.4.0 for
      Denial of Service (DoS) [medium severity][https://snyk.io/vuln/SNYK-JAVA-COMSQUAREUPOKIO-5773320]
```

akka - from s3_mock, test only, no fixed version
jetty - from lucene-replicator, I don't believe we use this functionality
okio - from grpc-okhttp, no fixed version

### CODE REVIEW

File: example-plugin/build.gradle [hunk 1]
```diff
@@ -32,7 +32,7 @@ dependencies {
     compileOnly "com.yelp.nrtsearch:server:${nrtsearchVersion}"
     testImplementation "com.yelp.nrtsearch:server:${nrtsearchVersion}"
     testImplementation("com.yelp.nrtsearch:server:${nrtsearchVersion}:tests")
-    testImplementation "junit:junit:4.13"
+    testImplementation "junit:junit:4.13.2"
```
Comment (vim345) [inline example-plugin/build.gradle:35]: Can you just apply the version here from the other file, where you set the junit version?

**Review by vim345 — APPROVED** [review]
Just a minor suggestion.

File: example-plugin/build.gradle (hunk 1 above)
Comment (aprudhomme) [reply example-plugin/build.gradle:35]: The problem is that `example-plugin` is structured as its own gradle project, so it does not have access to variables from the other project.

Comment (vim345) [reply example-plugin/build.gradle:35]: LGTM.

**Pull request merged**